entity-registry/src/main/java/com/linkedin/metadata/aspect/plugins/PluginFactory.java

```diff
@@ -48,17 +49,66 @@ public static PluginFactory merge(
       PluginFactory b,
       @Nullable
           BiFunction<PluginConfiguration, List<ClassLoader>, PluginFactory> pluginFactoryProvider) {
+
+    if (b.isEmpty()) {
+      return a;
+    }
+    if (a.isEmpty()) {
+      return b;
+    }
+
     PluginConfiguration mergedPluginConfig =
         PluginConfiguration.merge(a.pluginConfiguration, b.pluginConfiguration);
     List<ClassLoader> mergedClassLoaders =
         Stream.concat(a.getClassLoaders().stream(), b.getClassLoaders().stream())
             .collect(Collectors.toList());
 
-    if (pluginFactoryProvider != null) {
-      return pluginFactoryProvider.apply(mergedPluginConfig, mergedClassLoaders);
-    } else {
-      return PluginFactory.withCustomClasspath(mergedPluginConfig, mergedClassLoaders);
+    if (!a.hasLoadedPlugins() && !b.hasLoadedPlugins()) {
+      if (pluginFactoryProvider != null) {
+        return pluginFactoryProvider.apply(mergedPluginConfig, mergedClassLoaders);
+      } else {
+        return PluginFactory.withCustomClasspath(mergedPluginConfig, mergedClassLoaders);
```

Comment:
The Spring Plugin Factory should not hit this condition. The pluginFactoryProvider condition above was added to allow for the Spring factory to work. I'll see if I can write a check or throw an exception since it shouldn't be possible given how the Spring Plugin Factory is injected.